lib/Dialect/LLHD/Simulator/State.cpp

```diff
@@ -134,10 +134,12 @@ State::~State() {
 
   for (auto &entry : instances) {
     auto inst = entry.getValue();
-    if (!inst.isEntity && inst.procState) {
+    if (inst.procState) {
       std::free(inst.procState->inst);
       std::free(inst.procState->senses);
       std::free(inst.procState);
```

Comment:
Would it be possible to keep these in a `unique_ptr`, to also avoid leaking memory on exceptions (unlikely I know, but just for the sake of cleanliness)?

Reply:
I'm currently generating those through `malloc`s in the generated initialization function (as it depends on the process layout). I'm not sure if it would be possible to promote it to a `unique_ptr` or something similar?

Reply:
Yeah I think there's a constructor for `unique_ptr` that takes ownership of a bare pointer and deallocates in RAII style.

Reply:
True! I could promote the signal value pointers and local states pointers. I'm not sure if the internal pointers of `ProcState` (`inst` and `senses`) could be upgraded as easily though.

Reply:
Yeah that's a good point. Technically you can hold the `inst` and `senses` memory in a `std::unique_ptr`, in addition to having the pointers stored in `inst.procState->{inst,senses}`. But that's just nitpicking; I'm happy with how it is now :+1: We can always refactor/reiterate at a later stage.